src/node_internals.h

```diff
@@ -382,21 +382,20 @@ namespace heap {
 v8::Maybe<void> WriteSnapshot(Environment* env, const char* filename);
 }
 
+/**
+ * Convenient RAII helper to mark a duration of work on a thread.
+ */
 class TraceEventScope {
  public:
-  TraceEventScope(const char* category,
-                  const char* name,
-                  void* id) : category_(category), name_(name), id_(id) {
-    TRACE_EVENT_NESTABLE_ASYNC_BEGIN0(category_, name_, id_);
-  }
-  ~TraceEventScope() {
-    TRACE_EVENT_NESTABLE_ASYNC_END0(category_, name_, id_);
+  TraceEventScope(const char* category, const char* name)
```

Comment:
Isn't `TRACE_EVENT0(category, name)` already the same thing? Could we delete this class and use that macro instead?
https://github.com/nodejs/node/blob/3bc23f555a8501f5bb37a78cbe866102d02984b5/src/tracing/trace_event_common.h#L46-L51

Reply:
Yeah, you are right. Also, these TRACE_EVENT_* macros are not safe to be wrapped like this as they are using macros like __LINE__ and static variables. I've removed TraceEventScope.